src/pkg/retention/res/selectors/label/selector.go

```diff
@@ -38,10 +38,22 @@ type selector struct {
 	labels []string
 }
 
-// Select candidates by regular expressions
+// Select candidates by the labels
 func (s *selector) Select(artifacts []*res.Candidate) ([]*res.Candidate, error) {
-	// TODO: REPLACE SAMPLE CODE WITH REAL IMPLEMENTATION
-	return artifacts, nil
+	selected := make([]*res.Candidate, 0)
```

Comment:
`append` on a nil slice makes a new slice for you, so we don't have to initialize it here. We can also make it a named return parameter to drop this line.

Additionally, the `len` check on the input shouldn't be needed, since the for loop will exit immediately if there's nothing to loop over.

What about something like this?

```go
func (s *selector) Select(artifacts []*res.Candidate) (selected []*res.Candidate, err error) {
    for _, a := range artifacts {
        if isMatched(s.labels, a.Labels, s.decoration) {
            selected = append(selected, a)
        }
    }
}
```